File: FE/bookshop/src/js/data/voucher.js
const voucherColumns = [
  {
    field: "VoucherCode",
    nameField: "Mã khuyễn mãi",
    class: "table-align-text",
    width: "112px",
  },
  {
    field: "VoucherType",
    nameField: "Kiểu voucher",
    class: "table-align-text",
    width: "112px",
    type: "enum",
    enumType: "VOUCHER_TYPE"
  },
  {
    field: "AmountDiscount",
    nameField: "Số tiền giảm",
    class: "table-align-text",
    width: "112px",
    textAlign: "right",
    type: "money",
  },
  {
    field: "VoucherCondition",
    nameField: "Điều kiện sử dụng",
    class: "table-align-text",
    width: "112px",
    textAlign: "right",
    type: "money",
  },
  {
    field: "Quantity",
    nameField: "Số lượng",
    class: "table-align-text",
    width: "112px",
  },
  {
    field: "ExpiredDate",
    nameField: "Ngày hết hạn",
    class: "table-align-text",
    width: "112px",
    textAlign: "center",
    type: "datetime",
  },
  {
    field: "Description",
    nameField: "Mô tả",
    class: "table-align-text",
    width: "112px",
  },
];

export default voucherColumns;

File: BE/BookShopOnline/BookShopOnline.Core/Entitites/Voucher.cs
using BookShopOnline.Core.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookShopOnline.Core.Entitites
{
    public class Voucher : BaseEntity
    {
        public Guid VoucherId { get; set; }
        public string VoucherCode { get; set; }
        public VoucherType VoucherType { get; set; }
        public Decimal AmountDiscount { get; set; }
        public Decimal VoucherCondition { get; set; }
        public int Quantity { get; set; }
        public string? Description { get; set; }
        public DateTime ExpiredDate { get; set; }

    }
}
